editor/src/clj/editor/defold_project.clj

```diff
@@ -566,19 +569,18 @@
             (g/set-property resource-node :resource new-resource))))
 
       (du/measuring process-metrics :mark-deleted
-        (g/transact transaction-metrics
-          (for [node (:mark-deleted plan)]
-            (let [flaw (resource-io/file-not-found-error node nil :fatal (g/node-value node :resource))]
-              (g/mark-defective node flaw)))))
+        (let [basis (g/now)]
+          (g/transact transaction-metrics
+            (for [node-id (:mark-deleted plan)]
+              (let [flaw (resource-io/file-not-found-error node-id nil :fatal (resource-node/resource basis node-id))]
+                (g/mark-defective node-id flaw))))))
```

Comment:
Mostly indentation due to obtaining the `basis` outside of the loop, but we're also using `resource-node/resource` here to avoid creating a bunch of `evaluation-contexts`.